### PR TITLE
Support Bulk Online Query

### PR DESCRIPTION
**Pull request by rooftoofwoof**
- [x] serialization
- [x] deserialization
- [x] tests

Future Todos
- [ ] fix failing to read some has-many results
- [ ] see if it's possible to plop into result struct. Nah

### CODE REVIEW

File: params_online.go [hunk 1]
```diff
@@ -55,6 +55,8 @@ func (p OnlineQueryParamsComplete) WithStaleness(feature any, duration time.Dura
 	return p
 }
 
+// TO ARROW?
```
Comment (rooftoofwoof) [inline params_online.go:58]: Remove

File: models_private.go [hunk 2]
```diff
@@ -24,6 +25,21 @@ type onlineQueryResponseSerialized struct {
...
+type onlineQueryResultFeather struct {
+	HasData    bool
+	ScalarData *arrow.Record
+	GroupsData map[Fqn]*arrow.Record
+	Errors     []ServerError
+	Meta       *QueryMeta
+}
+
+type QueryName = string
+type Fqn = string
```
Comment (rooftoofwoof) [inline models_private.go:37]: privatize

File: validation.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package chalk
```
Comment (rooftoofwoof) [inline validation.go:1]: maybe make these functions go into the internal package

File: feather.go [hunk 4]
```diff
@@ -0,0 +1,417 @@
+package chalk
```
Comment (rooftoofwoof) [inline feather.go:1]: Move into internal package

**Review by emarx — APPROVED** [review]

**Pull request merged**